Modules/Loadable/Markups/VTKWidgets/vtkSlicerMarkupsWidgetRepresentation2D.cxx

```diff
@@ -182,6 +218,77 @@ void vtkSlicerMarkupsWidgetRepresentation2D::GetWorldToSliceCoordinates(const do
 }
 
 
+//----------------------------------------------------------------------
+void vtkSlicerMarkupsWidgetRepresentation2D::GetLabelTextXYOffsets(double backgroundVolumeCenterXY[2], double  slicePos[2], double leadersLineOffset[2])
```

Comment:
This method is way too overcomplicated. We don't need to deal with triangles, but coordinate geometry operations can be used instead. It could be as simple as something like this:

```
double leadLineDirectionVector[3];
vtkMath::Subtract(controlPoint, centerPoint, leadLineDirectionVector);
vtkMath::Normalize(leadLineDirectionVector);
double labelPoint[3] = 
  {
  controlPoint[0] + leadLineDirectionVector[0] * lineLength,
  controlPoint[1] + leadLineDirectionVector[1] * lineLength,
  controlPoint[2] + leadLineDirectionVector[2] * lineLength
  };
```